File: oo-workshop/mobprogrammingexercises/src/main/java/com/example/domainmodeling/mobprogramming/ShoppingCartValidator.java
package com.example.domainmodeling.mobprogramming;

import com.example.domainmodeling.mobprogramming.solution.InvalidCustomerException;

public class ShoppingCartValidator {

    public static boolean IsShoppingCartValid(final ShoppingCart shoppingCart) throws InvalidCustomerException {
        if (shoppingCart.getCustomer() == null) {
            throw new InvalidCustomerException();
        }
        return true;
    }
}

File: oo-workshop/tsexercises/src/domainmodeling/shoppingCartValidator.ts
import {ShoppingCart} from './shoppingCart'
import {InvalidCustomerError} from './invalidCustomerError'

export class ShoppingCartValidator {

    public static IsShoppingCartValid(shoppingCart: ShoppingCart): boolean | InvalidCustomerError {
        if (shoppingCart.customer == null) {
            return new InvalidCustomerError()
        }
        return true
    }
}